expertsystem/ui/__init__.py

```diff
@@ -459,3 +465,26 @@ def _particle_with_spin_projection_to_dict(
         if item["Type"] == "Spin":
             item["Projection"] = spin_projection
     return output
+
+
+def get_intermediate_state_names(
+    solutions: List[StateTransitionGraph],
+) -> Set[str]:
+    """Extract the names of the intermediate states in the solutions."""
+    intermediate_states = set()
+    for graph in solutions:
+        for edge_id in graph.get_intermediate_state_edges():
+            edge_property = graph.edge_props[edge_id]
+            if isinstance(edge_property, dict):
+                intermediate_states.add(edge_property["Name"])
```

Comment:
we have to remember and clean up this one at the end. I would almost prefer if currently just the dict version is implemented (as thats whats currently returned by the solvers). Then once that is fixed its impossible to miss this change

Reply:
That's a good point. I'm working on moving this function to `state.properties`, but now I noticed that I messed up something in #269: the idea was to have that PR split `particles` into `nested_dicts`, `state.properties` and `state.combinatorics`... Follow-up soon

Reply:
Btw, there are more of these 'backward compatible' functions, e.g. the ones introduced in #262.

Reply:
Yes, if both cases are needed, then its of course unavoidable. I don't know the best way around this. My idea is to just spot those old unused cases via code coverage